scripts/lib/downloadImages.ts

```diff
@@ -17,23 +17,44 @@ import { createWriteStream } from "node:fs";
 import { finished } from "stream/promises";
 import { Readable } from "stream";
 import pMap from "p-map";
+import { startWebServer, closeWebServer } from "../lib/webServer";
 
 export async function downloadImages(
   images: Array<{ src: string; dest: string }>,
+  imagesPath: string,
 ) {
-  await pMap(
-    images,
-    async (img) => {
-      if (await pathExists(img.dest)) return;
-      const response = await fetch(img.src);
-      if (response.ok) {
-        await mkdirp(dirname(img.dest));
-        const stream = createWriteStream(img.dest);
-        await finished(Readable.fromWeb(response.body as any).pipe(stream));
-      } else {
-        console.log(`Error downloading ${img.src} to ${img.dest}`);
-      }
-    },
-    { concurrency: 4 },
+  const missingImagesResults = await Promise.all(
+    images.map(async (img) => {
+      const exists = await pathExists(img.dest);
+      return exists ? null : img;
+    }),
   );
+
+  const missingImages = missingImagesResults.filter(
+    (img): img is { src: string; dest: string } => img !== null,
+  );
```

Comment:
Does it work to instead type-cast the overall `missingImages` variable? Because the current type hint is a lie. The `img` argument can be `null`! It's only after the `filter` that we're confident there are no more `null`s.

Reply:
I tried it but I got this error in vs code trying with the types `{ src: string; dest: string; }[] ` and `Array<{ src: string; dest: string; }>` : Type 'null' is not assignable to type '{ src: string; dest: string; }

Reply:
Using flatMap we have a similar issue to when I used just filter, here there's an example: https://stackoverflow.com/a/74959389. As suggested in that answer, I used map and flat separately